beacon_node/lighthouse_network/src/discovery/mod.rs

```diff
@@ -1036,12 +1038,107 @@ impl<TSpec: EthSpec> NetworkBehaviour for Discovery<TSpec> {
+                    };
+
+                    let mut should_update_ip: bool = true;
+
+                    /* we cannot violate the ENR invariant */
+                    if let IpAddr::V4(ip4) = ip {
+                        if !is_global(ip4) {
+                            debug!(self.log, "Refusing to update ENR IP (unreachable)"; "ip" => ?ip);
+                            should_update_ip = false;
+                        }
+                    }
+
+                    let port: u16 = match transport {
+                        Protocol::Tcp(port) => port,
+                        Protocol::Udp(port) => port,
+                        _ => {
+                            crit!(self.log, "Attempt to update discovery with unsupported transport (this should never occur)"; "transport" => ?transport);
+                            return;
+                        }
+                    };
+
+                    let transport_is_tcp: bool = matches!(transport, Protocol::Tcp(_));
+
+                    if should_update_ip {
+                        let sock: SocketAddr = SocketAddr::new(ip, port);
+
+                        self.discv5.update_local_enr_socket(sock, transport_is_tcp);
+                        *self.network_globals.local_enr.write() = self.discv5.local_enr();
+                    } else {
+                        match ip {
+                            IpAddr::V4(_) => {
+                                if transport_is_tcp {
+                                    insert_into_enr("tcp4", &port);
+                                } else {
+                                    insert_into_enr("udp4", &port);
+                                }
+                            }
+                            IpAddr::V6(_) => {
+                                if transport_is_tcp {
+                                    insert_into_enr("tcp6", &port);
+                                } else {
+                                    insert_into_enr("udp6", &port);
+                                }
+                            }
+                        }
+
+                        *self.network_globals.local_enr.write() = self.discv5.local_enr();
+                    }
+
+                    let local_enr: Enr = self.discv5.local_enr();
+                    info!(self.log, "Updated local ENR"; "enr" => local_enr.to_base64(), "seq" => local_enr.seq(), "id"=> %local_enr.node_id(),
+                        "ip4" => ?local_enr.ip4(), "udp4"=> ?local_enr.udp4(), "tcp4" => ?local_enr.tcp4(), "tcp6" => ?local_enr.tcp6(), "udp6" => ?local_enr.udp6());
+                    enr::save_enr_to_disk(Path::new(&self.enr_dir), &self.local_enr(), &self.log);
+                };
+
+                match addr.iter().nth(1) {
+                    Some(Protocol::Tcp(port)) => update_enr(ip, Protocol::Tcp(port)),
+                    Some(Protocol::Udp(port)) => update_enr(ip, Protocol::Udp(port)),
+                    Some(Protocol::Quic) => update_enr(ip, Protocol::Quic),
```

Comment:
I think we might be using QuicV1. Should double check this. 

In either case a simple run should show this debug error.

Reply:
> I think we might be using QuicV1.

We are; good catch.

> In either case a simple run should show this debug error.

This will almost never occur (see [above](https://github.com/sigp/lighthouse/pull/4700#discussion_r1331245019) for context)